Zwei Patches, um ICMP-Nachrichten (a) sicherer zu machen und (b) in einem
bestimmten Fall nicht zu senden.

Notwendig.

diff -rub --unidir /pub/src/linux/kernel/linux/net/inet/icmp.c ./net/inet/icmp.c
--- /pub/src/linux/kernel/linux/net/inet/icmp.c	Mon Mar  6 10:22:08 1995
+++ ./net/inet/icmp.c	Fri Mar 31 09:26:02 1995
@@ -210,7 +210,7 @@
 	 */
 	 
 	len = dev->hard_header_len + sizeof(struct iphdr) + sizeof(struct icmphdr) +
-		sizeof(struct iphdr) + 8;	/* amount of header to return */
+		sizeof(struct iphdr) + 32;	/* amount of header to return */
 	   
 	skb = (struct sk_buff *) alloc_skb(len, GFP_ATOMIC);
 	if (skb == NULL) 
diff -rub --unidir /pub/src/linux/kernel/linux/net/inet/ip.c ./net/inet/ip.c
--- /pub/src/linux/kernel/linux/net/inet/ip.c	Sat Mar 18 06:06:09 1995
+++ ./net/inet/ip.c	Thu Mar 23 17:23:27 1995
@@ -1369,9 +1371,11 @@
 	if (dev == dev2)
 		return;
 #else
+#if 0  /* dangerous, will send nonsense to remote hosts */
 	if (dev == dev2)
 		icmp_send(skb, ICMP_REDIRECT, ICMP_REDIR_HOST, raddr, dev);
 #endif		
+#endif		
 
 	/*
 	 * We now allocate a new buffer, and copy the datagram into it.
